api/python/provisioner/commands/validate_setup.py

```diff
@@ -0,0 +1,201 @@
+                f"{missing_params} cannot be left empty or blank.. "
+                "These are mandatory to configure the setup."
+            )
+            raise ValueError(f"Mandatory param missing {missing_params}")
+
+
+@attr.s(auto_attribs=True)
+class StorageNodeParamsValidation:
+    hostname: str = StorageEnclosureParams.hostname
+    _optional_param = []
+
+    def __attrs_post_init__(self):
+        params = attr.asdict(self)
+        missing_params = []
+        for param, value in params.items():
+            if value == UNCHANGED and param not in self._optional_param:
+                missing_params.append(param)
+        if len(missing_params) > 0:
+            logger.error(
+                f"{missing_params} cannot be left empty or blank.. "
+                "These are mandatory to configure the setup."
+            )
+            raise ValueError(f"Mandatory param missing {missing_params}")
+
+
+@attr.s(auto_attribs=True)
+class NodeNetworkParamsValidation:
+    search_domains: str = NetworkParams.search_domains
+    dns_servers: str = NetworkParams.dns_servers
+    cluster_id: str = NodeNetworkParams.cluster_id
+    bmc_user: str = NodeNetworkParams.bmc_user
+    bmc_secret: str = NodeNetworkParams.bmc_secret
+    network_data_interfaces: str = NodeNetworkParams.network_data_interfaces
+    network_data_netmask: str = NodeNetworkParams.network_data_netmask
+    network_data_gateway: str = NodeNetworkParams.network_data_gateway
+    network_mgmt_interfaces: List = NodeNetworkParams.network_mgmt_interfaces
+    network_mgmt_netmask: str = NodeNetworkParams.network_mgmt_netmask
+    network_mgmt_gateway: str = NodeNetworkParams.network_mgmt_gateway
+    pvt_ip_addr: str = NodeNetworkParams.pvt_ip_addr
+    storage_metadata_device: str = StorageEnclosureParams.storage_metadata_device
+    storage_data_devices: str = StorageEnclosureParams.storage_data_devices
+
+
+    _optional_param = [
+        'search_domains',
+        'dns_servers',
+        'cluster_id',
+        'network_data_netmask',
+        'network_data_gateway',
+        'pvt_ip_addr',
+        'network_mgmt_interfaces',
+        'network_mgmt_netmask',
+        'network_mgmt_gateway'
+    ]
+
+    def __attrs_post_init__(self):
+        params = attr.asdict(self)
+        missing_params = []
+        for param, value in params.items():
+            if value == UNCHANGED and param not in self._optional_param:
+                missing_params.append(param)
+        if len(missing_params) > 0:
+            logger.error(
+                f"{missing_params} cannot be left empty or blank.. "
+                "These are mandatory to configure the setup."
+            )
+            raise ValueError(f"Mandatory param missing {missing_params}")
+
+
+@attr.s(auto_attribs=True)
+class NodeParamsValidation:
+    hostname: str = NodeParams.hostname
+    is_primary: str = NodeParams.is_primary
+    roles: str = NodeParams.roles
+    bmc_ip: str = NodeParams.bmc_ip
```

Comment:
bmc_ip should be optional as for HW we get this directly no need provide them in config.ini